### PR TITLE
Major Updates

### PR DESCRIPTION
**Pull request by ehsteve**

### CODE REVIEW

**Comment by samaloney** [comment]
Wow this is a monster - finally getting around to looking at this sorry for the _long_ delay

**Review by samaloney — COMMENTED** [review]
Looks good, few small issues.

I wonder if a small dev guide is needed as I wasn't able to install using the traditional `pip install -e [dev]` I know you switched to poetry but I don't know what the equivalent command is?

Also both in GitHub action and locally there are errors worryingly command like `plt.plot(energy, mat.absorption(energy), label=mat.name)` are failing as ` mat.absorption(energy)` is returning a scaler.

Also maybe I just can't see it but I can't find a link to a built version of the docs?

**Comment by samaloney** [comment]
I tried to push a fix for the error but don't have permission, you can see the change in this diff between my fork and the branch for this PR https://github.com/ehsteve/roentgen/compare/poetry...samaloney:roentgen:poetry

**Comment by ehsteve** [comment]
Thanks @samaloney! The docs for this PR are hosted https://roentgen.readthedocs.io/en/poetry/index.html

File: roentgen/absorption/test.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+m = Material({"Si": 1, "He": 1}, 5 * u.m)
+
+stack1 = Material("Ge", 500 * u.micron) + Material("cdte", 100 * u.micron)
+stack2 = Material("Si", 500 * u.micron) + Material("Al", 500 * u.micron)
+
+total_stack = stack1 + stack2
+
+print(m.density)
+d = util.get_material_density("Si") * 0.5 + util.get_material_density("He") * 0.5
+print(d == m.density)
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply roentgen/absorption/test.py:15]: No. I accidentally committed this file. :(

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,128 @@
+[tool.poetry]
+name = "roentgen"
+version = "1.1.0"
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply pyproject.toml:3]: Good catch. fixed.

File: docs/conf.py [hunk 3]
```diff
@@ -79,25 +80,19 @@
...
+#graphviz_output_format = "svg"
+
+#graphviz_dot_args = [
+#    '-Nfontsize=10',
+#    '-Nfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
+#    '-Efontsize=10',
+#    '-Efontname=Helvetica Neue, Helvetica, Arial, sans-serif',
+#    '-Gfontsize=10',
+#    '-Gfontname=Helvetica Neue, Helvetica, Arial, sans-serif'
+#]
```
[the remark this answers is not in the document]
Comment (ehsteve) [reply docs/conf.py]: > Looks good, few small issues.
> 
> I wonder if a small dev guide is needed as I wasn't able to install using the traditional `pip install -e [dev]` I know you switched to poetry but I don't know what the equivalent command is?

Good point! It is simple `poetry install --with dev,docs,gui`. I'll add it to the documentation and README.

**Comment by ehsteve** [comment]
> Also both in GitHub action and locally there are errors worryingly command like `plt.plot(energy, mat.absorption(energy), label=mat.name)` are failing as ` mat.absorption(energy)` is returning a scale

I think this had to do with that error you found. I don't see such errors currently and the documentation plots seem fine.

File: docs/conf.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehsteve) [reply docs/conf.py]: I was getting graphviz errors and so got rid of those settings but they have since been resolved so removing the commenting.

**Pull request merged**